core/services/ocr2/plugins/ocr2keeper/evm20/registry.go

```diff
@@ -0,0 +1,822 @@
+package evm
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	coreTypes "github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/patrickmn/go-cache"
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg"
+	"go.uber.org/multierr"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/client"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/logpoller"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/generated"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/generated/keeper_registry_wrapper2_0"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/generated/mercury_lookup_compatible_interface"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/ocr2/models"
+	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+const (
+	// DefaultUpkeepExpiration decides how long an upkeep info will be valid for. after it expires, a getUpkeepInfo
+	// call will be made to the registry to obtain the most recent upkeep info and refresh this cache.
+	DefaultUpkeepExpiration = 10 * time.Minute
+	// DefaultCooldownExpiration decides how long a Mercury upkeep will be put in cool down for the first time. within
+	// 10 minutes, subsequent failures will result in double amount of cool down period.
+	DefaultCooldownExpiration = 5 * time.Second
+	// DefaultApiErrExpiration decides a running sum of total errors of an upkeep in this 10 minutes window. it is used
+	// to decide how long the cool down period will be.
+	DefaultApiErrExpiration = 10 * time.Minute
+	// CleanupInterval decides when the expired items in cache will be deleted.
+	CleanupInterval = 15 * time.Minute
```

Comment:
ideally we could control these by tweaking the offchainconfig set on the registry, so that in case we need to change something, we don't have to ship a new version of the node... how hard would that be? I don't think it's a must do, but good to understand if these values are a 1 or 2 way door.

Reply:
Sounds good, but this change is not related to 2.1 so let's do it separately